deeplake/util/keys.py

```diff
@@ -183,15 +184,36 @@ def get_sequence_encoder_key(key: str, commit_id: str) -> str:
 
 
 def dataset_exists(storage) -> bool:
+    """
+    Returns true if a dataset exists at the given location.
+    NOTE: This does not verify if it is a VALID dataset, only that it exists and is likely a deeplake directory.
+    To verify the content, use :func:`dataset_valid`
+    """
     try:
-        storage[get_dataset_meta_key(FIRST_COMMIT_ID)]
-        return True
+        return (
+                get_dataset_meta_key(FIRST_COMMIT_ID) in storage
+                or get_version_control_info_key() in storage
+        )
     except S3GetAccessError as err:
         raise AuthorizationException("The dataset storage cannot be accessed") from err
-    except (KeyError, S3GetError) as err:
+    except S3GetError:
         return False
 
 
+def dataset_validate(storage) -> None:
+    """
+    Checks the structure of the dataset and throws a descriptive DatasetCorruptError for any problems.
+    """
+    try:
+        for file in [get_dataset_meta_key(FIRST_COMMIT_ID), get_version_control_info_key()]:
```

Comment:
seems you are checking just version control information. Do we need to check tensor and their information?

Reply:
I ended up just taking this function out. So need to worry anymore